clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -993,16 +1019,18 @@ private long sendEligibleCalls(long now) {
                     continue;
                 }
                 Call call = calls.remove(0);
-                int timeoutMs = calcTimeoutMsRemainingAsInt(now, call.deadlineMs);
+                int requestTimeoutMs = Math.min(KafkaAdminClient.this.requestTimeoutMs,
+                        calcTimeoutMsRemainingAsInt(now, call.deadlineMs));
```

Comment:
There are a few other calls to `calcTimeoutMsRemainingAsInt`, do we need to consider the overall request timeout there as well?

Reply:
I looked at the other callers and it seems they are just checking if the deadline has been reached. We have to do the `min` here since the request timeout may be smaller than the remaining time before the deadline.